src/md_lammps.c

```diff
@@ -473,4 +474,527 @@ static md_molecule_loader_i lammps_api = {
+				//Use the first complete coord combo. x,y,z,xs,ys picks x,y,z and x,y,xs,ys,zs picks xs,ys,zs
+				if (
+					str_equal_cstr(tokens[i], "x") && str_equal_cstr(tokens[i + 1], "y") && str_equal_cstr(tokens[i + 2], "z") ||
+					str_equal_cstr(tokens[i], "xs") && str_equal_cstr(tokens[i + 1], "ys") && str_equal_cstr(tokens[i + 2], "zs") ||
+					str_equal_cstr(tokens[i], "xu") && str_equal_cstr(tokens[i + 1], "yu") && str_equal_cstr(tokens[i + 2], "zu") ||
+					str_equal_cstr(tokens[i], "xs") && str_equal_cstr(tokens[i + 1], "ys") && str_equal_cstr(tokens[i + 2], "zs")) {
+					coord_type = tokens[i];
+					coord_start = i - 2; //Skip ITEM: ATOMS tokens
+					break;
+				}
+			}
+			if (coord_start == -1) {
+				MD_LOG_ERROR("Could not parse coord definition in ITEM: ATOMS line");
+				return false;
+			}
+			//step = (int32_t)parse_int(str_substr(line, 10, 4));
+		}
+	}
+
+	//Extract coordinates and scale them if the type requires it. We leave unwrapped coordinates as is
+	int64_t i = 0;
+	while (str_extract_line(&line, &str) && i < traj_data->header.num_atoms) {
+		num_tokens = extract_tokens(tokens, 32, &line);
+		if (coord_start + 2 + 1 > num_tokens) {
+			MD_LOG_ERROR("Not enough tokens in atom line");
+			return false;
+		}
+		if (str_equal_cstr(coord_type, "x") || str_equal_cstr(coord_type, "xu")) { //Unscaled coordinates
+			x[i] = (float)parse_float(tokens[coord_start]);
+			y[i] = (float)parse_float(tokens[coord_start + 1]);
+			z[i] = (float)parse_float(tokens[coord_start + 2]);
+		}
+		else if (str_equal_cstr(coord_type, "xs") || str_equal_cstr(coord_type, "xsu")) { //Scaled coordinates
+			mat3_t unit_cell = traj_data->unit_cell.basis;
+			vec3_t scale_vec = {
+				(float)parse_float(tokens[coord_start]),
+				(float)parse_float(tokens[coord_start + 1]),
+				(float)parse_float(tokens[coord_start + 2]),
+			};
+			vec3_t result_vec = mat3_mul_vec3(unit_cell, scale_vec);
+			x[i] = result_vec.x;
+			y[i] = result_vec.y;
+			z[i] = result_vec.z;
+		}
+		i++;
+	}
+
+	if (header) {
+		header->num_atoms = traj_data->header.num_atoms;
+		header->index = frame_idx;
+		header->timestamp = (double)frame_idx;
+		header->unit_cell = traj_data->unit_cell;
+	}
+
+	return true;
+}
+
+static bool md_lammps_trajectory_parse(int64_t* num_atoms, md_unit_cell_t* unit_cell, int64_t* num_frame_offsets, md_array(int64_t)* frame_offsets, md_array(int64_t)* frame_times, md_buffered_reader_t* reader, struct md_allocator_i* alloc) {
+	ASSERT(reader);
+	ASSERT(alloc);
+	str_t line;
+	str_t tokens[6];
+	int64_t num_frames = 0;
+
+
+	//Setup frames and first frame
+	md_buffered_reader_skip_line(reader);
+	md_buffered_reader_extract_line(&line, reader);
+	md_array_push(*frame_times, parse_int(line), alloc);
+	num_frames++;
+
+
+	//Parse num of atoms
+	{
+		md_buffered_reader_skip_line(reader);
+		md_buffered_reader_extract_line(&line, reader); //Read num of atoms
+		*num_atoms = parse_int(line);
+	}
+
+	//Parse unit_cell definition
+	{
```

Comment:
Changed how this works so that any order can be handled now, as we are checking values in the main while loop instead. Uses bools such as box_bound_found to avoid multiple checks.